### PR TITLE
`Rewind/ActionButtons`: refactor tests to `@testing-library/react`

### PR DESCRIPTION
**Pull request by flootr**
#### Proposed Changes

* `Rewind/ActionButtons`: refactor tests to `@testing-library/react`

#### Testing Instructions

* Verify unit tests still pass

Related to #63409

### CODE REVIEW

File: client/components/jetpack/daily-backup-status/test/action-buttons.jsx [hunk 1]
```diff
@@ -34,69 +26,79 @@ describe( 'ActionButtons', () => {
...
-		const wrapper = render( <ActionButtons rewindId="test" /> );
-		const downloadButton = wrapper.find( '.daily-backup-status__download-button' ).hostNodes();
-
-		expect( downloadButton.prop( 'href' ) ).toBeTruthy();
-		expect( downloadButton.prop( 'disabled' ) ).toBeFalsy();
+		const rewindId = 'test';
+		render( <ActionButtons rewindId={ rewindId } /> );
+		const downloadButton = screen.getByRole( 'link', { name: /download/i } );
+
+		expect( downloadButton ).toHaveProperty(
```
Comment (tyxla) [inline client/components/jetpack/daily-backup-status/test/action-buttons.jsx]: Did you mean:

```suggestion
		expect( downloadButton ).toHaveAttribute(
```

File: client/components/jetpack/daily-backup-status/test/action-buttons.jsx [hunk 2]
```diff
@@ -34,69 +26,79 @@ describe( 'ActionButtons', () => {
...
+		const rewindId = 'test';
 
-		const wrapper = render( <ActionButtons rewindId="test" /> );
-		const restoreButton = wrapper.find( '.daily-backup-status__restore-button' ).hostNodes();
+		render( <ActionButtons rewindId={ rewindId } /> );
+		const restoreButton = screen.getByRole( 'link', { name: /restore/i } );
 
-		expect( restoreButton.prop( 'href' ) ).toBeTruthy();
-		expect( restoreButton.prop( 'disabled' ) ).toBeFalsy();
+		expect( restoreButton ).toHaveProperty(
```
Comment (tyxla) [inline client/components/jetpack/daily-backup-status/test/action-buttons.jsx]: Did you mean:

```suggestion
		expect( restoreButton ).toHaveAttribute(
```

**Review by tyxla — APPROVED** [review]
Besides a couple of minor suggestions, this looks ace 🚀

File: client/components/jetpack/daily-backup-status/test/action-buttons.jsx (hunk 1 above)
Comment (flootr) [reply client/components/jetpack/daily-backup-status/test/action-buttons.jsx]: Yes, good catch. Thanks!

**Pull request merged**